### PR TITLE
Additional Migration Work

### PR DESCRIPTION
**Pull request by gene-redpanda**
Renamed MigrationAction's elements, added a MigrationStatus and added functions to get them from strings. Also added some tests and fixed some bugs.

### CODE REVIEW

File: rpadmin/api_mount.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+	"context"
+	"net/http"
+)
+
+const (
+	baseMountEndpoint   = "/v1/topics/mount"
+	baseUnmountEndpoint = "/v1/topics/unmount"
+)
```
[the remark this answers is not in the document]
Comment (gene-redpanda) [reply rpadmin/api_mount.go:11]: Mostly just habit 🤷‍♀️

File: rpadmin/api_mount.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+// MigrationInfo represents the underlying migration information returned by mount and unmount operations
+type MigrationInfo struct {
+	ID int `json:"id"`
+}
+
+// MountTopics mounts topics according to the provided configuration
+func (a *AdminAPI) MountTopics(ctx context.Context, config MountConfiguration) (MigrationInfo, error) {
+	var response MigrationInfo
+	err := a.sendAny(ctx, http.MethodPost, baseMountEndpoint, config, &response)
```
[the remark this answers is not in the document]
Comment (gene-redpanda) [reply rpadmin/api_mount.go:26]: My understanding is that any is valid but I'm validating that part still

File: rpadmin/api_mount.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	return response, err
+}
+
+// UnmountTopics unmounts the provided list of topics
+func (a *AdminAPI) UnmountTopics(ctx context.Context, topics []Topic) (MigrationInfo, error) {
```
[the remark this answers is not in the document]
Comment (gene-redpanda) [reply rpadmin/api_mount.go]: Done, I wish I had thought of that earlier 😅

File: rpadmin/api_migration.go [hunk 4]
```diff
@@ -60,68 +62,39 @@ func (a *AdminAPI) ListMigrations(ctx context.Context) ([]MigrationState, error)
...
 }
 
 // InboundTopic represents an inbound migration topic
 type InboundTopic struct {
-	SourceTopic Topic  `json:"source_topic"`
-	Alias       *Topic `json:"alias,omitempty"`
-	Location    string `json:"location,omitempty"`
+	SourceTopic NamespacedTopic  `json:"source_topic"`
+	Alias       *NamespacedTopic `json:"alias,omitempty"`
+	Location    string           `json:"location,omitempty"`
```
Comment (weeco) [inline rpadmin/api_migration.go:97]: I think `location` doesn't exist (anymore?). If seen this as well somewhere, but here's the schema I'm referring to

**Review by weeco — APPROVED** [review]

**Review by bojand — APPROVED** [review]

**Pull request merged**